src/liballoc/vec.rs

```diff
@@ -1754,7 +1819,6 @@ impl<T> IntoIterator for Vec<T> {
     fn into_iter(mut self) -> IntoIter<T> {
         unsafe {
             let begin = self.as_mut_ptr();
-            assume(!begin.is_null());
```

Comment:
`as_mut_ptr` now contains an `assume` so I think I can remove this? Unfortunately whoever added this did not leave a comment, because it seemed redundant before already.